application/src/main/scala/com/gu/deskcomexporttool/DeskComClient.scala

```diff
@@ -0,0 +1,76 @@
+package com.gu.deskcomexporttool
+
+import java.util.Base64
+
+import cats.data.EitherT
+import cats.instances.future._
+import io.circe.derivation._
+import io.circe.parser._
+import io.circe.{Decoder, derivation}
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait DeskComClient {
+  def getAllInteractions(page: Int, pageSize: Int): EitherT[Future, DeskComApiError, List[Interaction]]
+
+  def close(): Unit
+}
+
+object DeskComClient {
+  private val log = LoggerFactory.getLogger(this.getClass)
+
+  def apply(config: DeskComApiConfig, httpClient: HttpClient)(implicit ec: ExecutionContext): DeskComClient = new DeskComClient() {
+    private val authHeader = HttpHeader(
+      "Authorization",
+      s"Basic ${Base64.getEncoder.encodeToString(s"${config.username}:${config.password}".getBytes("UTF-8"))}"
+    )
+
+    implicit val interactionDecoder: Decoder[Interaction] = deriveDecoder(derivation.renaming.snakeCase)
+    implicit val interactionEmbeddedDecoder: Decoder[GetInteractionsEmbedded] = deriveDecoder(derivation.renaming.snakeCase)
+    implicit val interactionResponseDecoder: Decoder[GetInteractionsResponse] = deriveDecoder(derivation.renaming.snakeCase)
+
+    override def getAllInteractions(page: Int, pageSize: Int): EitherT[Future, DeskComApiError, List[Interaction]] = {
+      for {
+        httpResponse <- httpClient.request(
+          HttpRequest(
+            "GET",
+            s"${config.baseUrl}/api/v2/interactions?page=$page&per_page=$pageSize&sort_field=created_at&sort_direction=asc",
+            List(authHeader)
+          )
+        ).leftMap(httpError => DeskComApiError(s"Request for interactions failed: $httpError"))
+        _ <- EitherT.fromEither(validateStatusCode(httpResponse.statusCode))
```

Comment:
Do we need this?

Reply:
validateStatusCode will cause a failure if it returns a left, if it returns a right we just ignore it and use the response from the previous step